### ISSUE TITLE
[a11y]: DataTable [dynamic] - expand/collapse control present in each row

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#14262]
### Package

@carbon/react

### Browser

FF & Safari & iPhone

### Package version

@carbon/react@1.32.0

### Automated testing tool and ruleset

Manual QA

### Assistive technology

JAWS & VO

### Description
Expand/collapse control present in each Row

The Show/Hide content is identified by SR even when it is collapsed

The Show/Hide content is identified by SR even when it is collapsed. Note: This issue is observed in "expansion" and Batch Expansion  variants as well.

### Recommended Fix
SR is not expected to identify the visually hidden content. Hide the content and make it available for SR once its expanded.

If the screen reader identifies the visually hidden content, it tends to mislead users who are blind. 

### WCAG 2.1 Violation

4.2.1 Roles, Names and Values

### Reproduction/example
Record `Number 66` in the below table:
https://airtable.com/shrFsaeHhImOkuVNR

https://react.carbondesignsystem.com/?path=/story/components-datatable-dynamic--default

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
I think fixing this may also fix https://github.com/carbon-design-system/carbon/issues/14261
